<launch>

    <!-- import joy button/axis mapping settings -->
    <rosparam file="$(find nhk2018_main_ros)/param/joy_assign_wired.yaml" command="load" />

    <!-- arg name="use_sim_time" default="false"/>
    <param name="use_sim_time" value="$(arg use_sim_time" /-->
    <arg name="use_joy" value="true" />
    
    <!-- base controller -->
    <node pkg="base_controller" type="base_controller" name="base_ctrl" output="screen">
       <param name="motor_max_acc" value="300.0" />
       <param name="motor_max_vel" value="200.0" />
       <param name="invert_x" value="false" />
       <param name="invert_y" value="true" />
       <param name="invert_z" value="false" />
       <remap from="motor_cmd_vel" to="base/motor_cmd_vel" />
    </node>
    
    <!-- joy node -->
    <node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
        <param name="dev" type="string" value="/dev/input/js0" />
        <param name="deadzone" value="0.12" />
    </node>
    
    <include file="$(find navigation)/launch/navigation.launch">
        <arg name="use_amcl" value="false" />
        <arg name="map_file" value="$(find navigation)/map/abu2018_map.yaml" />
        <arg name="odom_x_coeff" value="-1.0" />
        <arg name="odom_y_coeff" value="-1.0" />
        <arg name="odom_yaw_coeff" value="1.0" />
        <arg name="odom_swap_xy" value="false" />
        <arg name="useless_planner_conf_file" value="$(find navigation)/param/useless_planner_params_cr.yaml" />
    </include>

    <!-- tr_main -->
    <!-- node pkg="nhk2018_main_ros" type="tr_main_auto_1v" name="main" output="screen">
        <param name="tr" value="2" />
        <rosparam param="unchuck_thres">[450, 400, 144]</rosparam>
    </node -->
    
    <node pkg="nhk2018_main_ros" type="cr_main_semiauto_2v0" name="main" output="screen">
        <param name="amt_coeff" value="1.0" />
    </node>
    
    <!-- joy teleop -->
    <!--node pkg="base_controller" type="base_teleop_joy" name="base_teleop_joy"/-->
    
    <!-- joy node -->
    <!-- node respawn="true" pkg="joy" type="joy_node" name="joy_node" >
        <param name="dev" type="string" value="/dev/input/js0" />
        <param name="deadzone" value="0.12" />
    </node-->
    
    
    <!-- shutdown controller -->
    <!-- node pkg="base_controller" type="shutdown_controller_joy" name="shutdown_ctrl_joy" /-->
  
    <node pkg="nhk2018_main_ros" type="cr_can" name="cr_can" />
    
    <!-- usb_can -->
    <node pkg="nhk2018_main_ros" type="usb_can_node" name="usb_can_node">
    </node>
    
    
    
    <!-- hand itself -->
    <node pkg="rosserial_server" type="serial_node" name="hands">
<rosparam>
port: /dev/ttyUSB1
baud: 57600
require:
  publishers: [  ]
  subscribers: [ shutdown ]
</rosparam>
    </node>
    
    <node pkg="rosserial_python" type="message_info_service.py" name="rosserial_message_info" />
</launch>
